### ISSUE TITLE
[Process Improvement] Give issues a time estimate

### ISSUE THREAD

**Issue by simontegg** [open-app/holodex#60]
Time estimates may be entirely inaccurate and sft. But may help communicate to others what may happen by when

**Comment by ahdinosaur** [comment]
how about a 'blocked' label?

**Comment by ahdinosaur** [comment]
:+1:

is your suggestion to write an estimate of the time within the issue ("i think this might be done by next Tuesday"), or weight the issue in waffle ("this issue is a 3")?

**Comment by simontegg** [comment]
The former. See #63.  A checkin frequency is included

**Comment by ahdinosaur** [comment]
sweet. as an aside, i reckon we should write these processes down somewhere in `./process`.

**Comment by bhaugen** [comment]
I agree with "when might this be done" rather than weight. Much more useful, and I find that I can meet that kind of estimate better than anything like "how much time or work will this take me".

**Comment by simontegg** [comment]
What is the time estimate on #56? We (I) also made a commitment to have all of enspiral's data in holodex by the end of the month (tomorow). Where is this ticket?

**Comment by ahdinosaur** [comment]
@simontegg https://github.com/open-app/holodex/issues/56 isn't something i feel comfortable estimating given the conditions. the ticket for enspiral's data is https://github.com/holodex/enspiral-data/issues/5, progress is in https://github.com/holodex/enspiral-data/tree/kumu.

**Comment by simontegg** [comment]
Ok. I don't understand what the conditions are and we can talk about this on Wednesday. I'm flagging that I still don't feel super about investing in Holodex, but I do have more clarity. Some conditions that I'm thinking about for me to contribute are:
- getting paid (I will work for below market rate)
- I value certainty, predictability and fairness. Regular value delivery creates more certainty and more predictability. Time estimates create more predictability. Sharing in chores (like data entry) creates more fairness.
- Given those values, I will scale any unpaid efforts to match the value I perceive being delivered from other team members if I have capacity.  Effectively this means doing #63 by when I said I would. I have capacity to do more but I don't feel good about doing more currently. 
- I won't lead out on comms  until I have observed us meeting minimum standards of reliability which I don't feel we currently meet. Others may lead out on comms and I might participate in followup though. 

The main way to support me working on Holodex is to make regular, predictable improvements to the product, other forms of user/customer value (comms, data entry, marketing site), management (problem identification, annotation, backlog grooming, testing suite) and process improvements. 

Some perspectives that inform this are: 

> Sometimes you’ll have to bet on big, resource intensive projects that won’t work. **Be careful of the morale toll on your team**. Not everyone has great tolerance for repeated failure. If you’re a founder, you probably are more used to this than your team. **One technique is to mix iterative work with riskier bigger bets to rebuild this energy**. But you should also try to not paint any feature as a silver bullet, and individual failure in good faith as a necessary cost of trying new things. **Your company is a portfolio bet.**
> https://medium.com/@gerstenzang/21-management-things-i-learned-at-imgur-7abb72bdf8bf
> 
> The saying goes, you don't want ninjas, you want ants. Ninjas are never there. Ants carry a hundred times their own weight and are great at working in teams. We're experts in helping the ants be happy and productive.
> -Peter Hintjens http://www.imatix.com/

**Comment by ahdinosaur** [comment]
sweet @simontegg, thanks for sharing. keen to talk about it on Wednesday.

**Comment by simontegg** [comment]
@ahdinosaur  In regard to our chat about Non Violent Communication today I have a few thoughts:
- I appreciate the value of this. I'll look into it more.
- I would prefer we gain consent before adopting a standard or philosophy of communication. Particularly if that standard assumes a high level of trust and emotional intimacy.
- It shouldn't be a condition to contributing.
- Neither of us are trained in NVC. Just an observation. 
- There's probably quite a bit to it. Does it work if we partially adopt some techniques? There are people in the network who have more experience in this. I'm going to seek other's perspectives and encourage you to do the same.
  
  I'm interested in how it works with communicating with external stakeholders. Ie. it may be true that expectations like emotions are the responsibility of the person concerned. However, stakeholders are unlikely to consent to a NVC ethos; are going to form expectations; and if we fail to meet these uncommunicated expectations, we could say "That's unfortunate for you. Your expectations are your responsibility. [try communicating them in a non-violent way]" but that's unlikely to go down well :)

This points to the value of proactive expectations management. I can agree that emotions and expectations are the person's responsibility. But I also know that proactive emotional management of others is good (being proactively nice to people to help them feel good)  (yes?). Does this also mean that proactive expectations management is valuable?

In summary, thinking out loud, we can create a virtuous cycle of trust and certainty through the following (in order of preference) :
1. strong outcome commitments "X will be delivered by Y time" that become true.
2. soft outcome commitments "I aim have X delivered by Y time, this might change", plus proactive expectations management "I'll keep stakeholders updated with the status and the expectations they should have, regularly, and as things change "
3. process commitments
